clang/test/ExtractAPI/inherited_availability.m

```diff
@@ -0,0 +1,149 @@
+// RUN: rm -rf %t
+// RUN: %clang_cc1 -extract-api --pretty-sgf --emit-sgf-symbol-labels-for-testing -triple arm64-apple-macosx \
+// RUN:   -x objective-c-header %s -o %t/output.symbols.json -verify
+
+
+// RUN: FileCheck %s --input-file %t/output.symbols.json --check-prefix A
+__attribute__((availability(macos, introduced=9.0, deprecated=12.0, obsoleted=20.0)))
+@interface A
```

Comment:
What's the expected availability information emitted when the enclosing entity doesn't have one defined? 
```
@interface A
@property int moreRestrictive __attribute__((availability(macos, introduced=10.0, deprecated=11.0, obsoleted=19.0)));
@end 
```

Reply:
It walks DeclContext's upwards to find one to compute the most restrictive information. If the are no other annotations for that platform walking upwards then it is just the one in the availability annotation. I realized that the existing availability test case doesn't cover this, thanks for pointing it out I will add a test to validate this behavior.